src/Modules/ASayModule/Services/ASayService.cs

```diff
@@ -0,0 +1,36 @@
+﻿using EvoSC.Common.Services.Attributes;
+using EvoSC.Common.Services.Models;
+using EvoSC.Manialinks.Interfaces;
+using EvoSC.Modules.Official.ASayModule.Interfaces;
+using Microsoft.Extensions.Logging;
+
+namespace EvoSC.Modules.Official.ASayModule.Services;
+
+[Service(LifeStyle = ServiceLifeStyle.Transient)]
+public class ASayService : IASayService
+{
+    private readonly ILogger<ASayService> _logger;
+    private readonly IManialinkManager _manialinkManager;
+
+    public ASayService(ILogger<ASayService> logger, IManialinkManager manialinkManager)
+    {
+        _logger = logger;
+        _manialinkManager = manialinkManager;
+    }
+
+    public async Task OnDisableAsync()
+    {
+        await _manialinkManager.HideManialinkAsync("ASayModule.Announcement");
+        _logger.LogInformation("ASayModule disabled.");
+    }
+
+    public async Task ShowAnnouncementAsync(string text)
+    {
+        await _manialinkManager.SendManialinkAsync("ASayModule.Announcement", new {text});
```

Comment:
I think it would be an idea to log who creates an ASay message in case a server admin wants to check who is misusing their rights.

Reply:
fixed